### PR TITLE
🐛Fix docking positioning bugs

### LINKED ISSUE #15403: Video sometimes gets hidden by browser UI
[as noted](https://github.com/ampproject/amphtml/pull/14961#issuecomment-388213491)

### PR DESCRIPTION
**Pull request by alanorozco**
Fixes #15403, #15402

(Also fixes an infinite timeout 😰)

### CODE REVIEW

File: src/service/video/docking.js [hunk 1]
```diff
@@ -1129,10 +1146,13 @@ export class VideoDocking {
     const {x, y} = pointerCoords(e);
 
     const offset = {x: 0, y: 0};
+    const {posX, posY} = this.currentlyDocked_;
```
Comment (nainar) [inline src/service/video/docking.js]: s/pos/currentlyDockedPos

File: src/service/video/docking.js [hunk 2]
```diff
@@ -600,11 +600,12 @@ export class VideoDocking {
 
   /**
    * Reconciliates the state of a docked or potentially dockable video when
-   * its position changes.
+   * something (viewport, position) changes.
```
Comment (nainar) [inline src/service/video/docking.js]: nit: s/"something (viewport, position)"/'viewport/position"

File: src/service/video/docking.js [hunk 3]
```diff
@@ -803,30 +809,41 @@ export class VideoDocking {
...
+  }
+
+  /**
+   * @param {!../../video-interface.VideoOrBaseElementDef} video
+   * @param {!RelativeX} posX
+   * @param {!RelativeY} posY
+   * @param {?number=} optStep
+   * @private
+   */
+  dock_(video, posX, posY, optStep = null) {
```
Comment (cvializ) [inline src/service/video/docking.js]: nit: the format for optional parameters used in most AMP code is `opt_variableName`

Comment (cvializ) [inline src/service/video/docking.js]: Does `optStep` need to be explicitly `null`, or does `undefined` work too? Parameters that are not passed are undefined by default, so the assignment below can be
```js
dock_(video, posX, posY, optStep) {
  const step =
      optStep != null ? optStep : this.calculateStep_(video.element);
  /*...*/
}
```

File: src/service/video/docking.js [hunk 4]
```diff
@@ -600,11 +600,12 @@ export class VideoDocking {
...
   /**
    * Reconciliates the state of a docked or potentially dockable video when
-   * its position changes.
+   * something (viewport, position) changes.
    * @param {!../../video-interface.VideoOrBaseElementDef} video
+   * @param {boolean=} onPositionChange
    * @private
    */
-  onPositionChange_(video) {
+  update_(video, onPositionChange = false) {
```
Comment (cvializ) [inline src/service/video/docking.js]: Using flag arguments is a little bit of an antipattern. Consider adding a separate function for the `onPositionChange == true` case.

https://martinfowler.com/bliki/FlagArgument.html

File: src/service/video/docking.js (hunk 1 above)
Comment (alanorozco) [reply src/service/video/docking.js]: `currentPosX` :)

File: src/service/video/docking.js (hunk 2 above)
Comment (alanorozco) [reply src/service/video/docking.js]: Done.

File: src/service/video/docking.js [hunk 5]
```diff
@@ -1398,9 +1424,7 @@ export class VideoDocking {
   // Using `vw` and `vh` to disambiguate with `width` and `height` when
   // destructuring.
   getViewportSize_() {
-    const viewportSize = this.viewport_.getSize();
```
[the remark this answers is not in the document]
Comment (alanorozco) [reply src/service/video/docking.js]: The reason for the abstraction is to take margins into account in the future.

File: src/service/video/docking.js (hunk 3 above)
Comment (alanorozco) [reply src/service/video/docking.js]: Done.

Comment (alanorozco) [reply src/service/video/docking.js]: Yes, undefined is supposed to work. Alternatively using `isFiniteNumber` for clarity.

File: src/service/video/docking.js (hunk 4 above)
Comment (alanorozco) [reply src/service/video/docking.js]: Done.

File: src/service/video/docking.js (hunk 3 above)
Comment (cvializ) [reply src/service/video/docking.js]: nice, good use of `isFiniteNumber`. I need to use that more

**Pull request merged**